bucket_map/src/bucket_map.rs

```diff
@@ -137,10 +135,17 @@ impl<T: Clone + Copy + Debug> BucketMap<T> {
         Some(self.buckets[ix].read().unwrap().as_ref()?.range(range))
     }
 
+    /// Get the Pubkeys for bucket `ix`
+    pub fn keys(&self, ix: usize) -> Option<Vec<Pubkey>> {
```

Comment:
I think we get rid of option here and just return vec![] if there is no bucket. Caller doesn't need to know bucket doesn't exist. That isn't a case in the way we intend to use it anyway. All buckets should be allocated.Same on range I think.